Support spellchecking in docs content

ptgott opened the pull request:
In gravitational/docs#261, we will add a script that checks the spelling of each version of the docs. This change edits one version of the docs content to support this, including:

- A cspell configuration file
- A new step in the GitHub Actions in the "Lint (Docs)" workflow that runs the spellcheck script we will add in `gravitational/docs`
- Fix mispellings so this passes the lint job. The mispellings are in a file that we generated automatically, but there are few enough of them, and we haven't merged the auto-generation script yet, that I think it makes sense to fix them in the generated file for now.

ptgott committed:
Support spellchecking in docs content

In gravitational/docs#261, we will add a script that checks the spelling
of each version of the docs. This change edits one version of the docs
content to support this, including:

- A cspell configuration file
- A new step in the GitHub Actions in the "Lint (Docs)" workflow that
  runs the spellcheck script we will add in `gravitational/docs`
- Fix mispellings so this passes the lint job. The mispellings are in a
  file that we generated automatically, but there are few enough of
  them, and we haven't merged the auto-generation script yet, that I
  think it makes sense to fix them in the generated file for now.

hatched commented on docs/cspell.json:
```diff
@@ -0,0 +1,699 @@
+    "version": "0.2",
+    // language - current active spelling language
+    "language": "en",
+    // words - list of words to be always considered correct
```
Does cspell support comments like this? `json` doesn't support comments in the spec...

ptgott replied:
This is the config Steven has been using—it looks like `cspell` ignores the comments. I can still remove them to prevent weirdness in editors.

hatched commented on docs/cspell.json:
```diff
@@ -0,0 +1,699 @@
+"elbv",
+"elbz",
+"elif",
+"emtpy",
```
I think this is probably a mispelling that should be updated in the docs?

stevenGravy replied:
This is from the terraform code blocks, docs/pages/deploy-a-cluster/deployments/aws-terraform.mdx

```code
if [[ "${INSTANCE_TYPE}" == "auth" ]]; then
    export SERVER_IP=$(aws ec2 describe-instances --filters "Name=tag:TeleportCluster,Values=${TF_VAR_cluster_name}" "Name=tag:TeleportRole,Values=auth" --query "Reservations[${INSTANCE_ID}].Instances[*].PrivateIpAddress" --output text)
    echo "Auth ${INSTANCE_ID} IP: ${SERVER_IP}"
elif [[ "${INSTANCE_TYPE}" == "proxy" ]]; then
    export SERVER_IP=$(aws ec2 describe-instances --filters "Name=tag:TeleportCluster,Values=${TF_VAR_cluster_name}" "Name=tag:TeleportRole,Values=proxy" --query "Reservations[${INSTANCE_ID}].Instances[*].PrivateIpAddress" --output text)
    echo "Proxy ${INSTANCE_ID} IP: ${SERVER_IP}"
elif [[ "${INSTANCE_TYPE}" == "node" ]]; then
    export SERVER_IP=$(aws ec2 describe-instances --filters "Name=tag:TeleportCluster,Values=${TF_VAR_cluster_name}" "Name=tag:TeleportRole,Values=node" --query "Reservations[*].Instances[*].PrivateIpAddress" --output text)
    echo "Node IP: ${SERVER_IP}"
fi
```

stevenGravy replied:
Yes, ./pages/management/guides/joining-nodes-aws-iam.mdx:create and attach an emtpy role with no attached policies.

hatched commented on docs/cspell.json:
```diff
@@ -0,0 +1,699 @@
+"dbname",
+"DBSIZE",
+"dbuser",
+"DEBU",
```
Probably a real mispelling, should be `DEBUG`

stevenGravy replied:
Similar to ERRO we have output that has this

hatched commented on docs/cspell.json:
```diff
@@ -0,0 +1,699 @@
+    "language": "en",
+    // words - list of words to be always considered correct
+    "words": [
+"ANPAW",
```
This list should probably be alphabetical

hatched commented on docs/cspell.json:
```diff
@@ -0,0 +1,699 @@
+"yourcompany",
+"yourtenant",
+"Ypga",
+"Yubi",
```
I don't see this anywhere in the docs...

hatched commented on docs/cspell.json:
```diff
@@ -0,0 +1,699 @@
+    ],
+    // flagWords - list of words to be always considered incorrect
+    // This is useful for offensive words and common spelling errors.
+    // For example "hte" should be "the"
```
comments in json.

hatched commented on docs/cspell.json:
```diff
@@ -0,0 +1,699 @@
+"FLUSHALL",
+"FLUSHDB",
+"FMPFM",
+"foregound",
```
that says "foreGOUND", missing the R 😁

ptgott committed:
Respond to PR feedback

- Remove misspellings from the ignore list
- Sort the ignore list (and format it via prettier)

hatched commented on docs/cspell.json:
```diff
@@ -0,0 +1,686 @@
+    "tenantname",
+    "testuser",
+    "thisisunsafe",
+    "thred",
```
The only instance of "thred" in the docs is "thredUP". Is there a config that maybe makes this check case insensitive? As with this addition a misspelling of "thred" for "thread" will be ignored.

ptgott replied:
I removed this entry in bbdc3421240f55fc14cb439a081efca6c3480852 and the spell checker didn't flag any misspellings.

ptgott commented:
@alexfornuto Just checking to see if you have any feedback on this

alexfornuto reviewed (approved):
What is it the kids say? I'm here for it!

ptgott commented:
This currently isn't working as expected because the `yarn spellcheck` command exits with a zero status if there's an error. It also is erroneously checking `content/<ver>` directories in the GHA workflow container, while the `gravitational/teleport` submodule lives in `content/docs`. 

I have a PR here to fix the issues: https://github.com/gravitational/docs/pull/264

Once that PR is merged, we can change the GHA workflow to run `yarn spellcheck` on the content directory where we mount the docs submodule.

ptgott committed:
Use the new yarn spellcheck command

ptgott committed:
Spelling fixes

stevenGravy committed:
Merge branch 'master' into paul.gottschling/2023-03-29-fix-spelling

stevenGravy committed:
spell fixes and add words to cspell.json
